coordinator/bridge/src/main/java/io/stargate/bridge/codec/VectorCodec.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.bridge.codec;
+
+import com.datastax.oss.driver.api.core.type.codec.TypeCodecs;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.stargate.bridge.grpc.Values;
+import io.stargate.bridge.proto.QueryOuterClass.Collection;
+import io.stargate.bridge.proto.QueryOuterClass.Value;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.Column.ColumnType;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class VectorCodec extends CompositeCodec {
+
+  @Override
+  public ByteBuffer encode(@NonNull Value value, @NonNull ColumnType type) {
+    if (value.hasNull()) return null;
+    if (!value.hasCollection()) {
+      throw new IllegalArgumentException("Expected collection of float type");
+    }
+    Collection collection = value.getCollection();
+    int elementCount = collection.getElementsCount();
+
+    ByteBuffer[] encodedElements = new ByteBuffer[elementCount];
```

Comment:
Seems very inefficient to construct separate `ByteBuffer` for every element, and then in the end create appended version. Instead of allocating one bigger `ByteBuffer` and appending directly into that. 
Would it be possible to change this? (I am guessing code is copied from driver or so).

Reply:
Not sure how to avoid multiple ByteBuffer creation, we could avoid creating the array. But would keep it like this so the code will look similar to what's done in driver code [here](https://github.com/datastax/java-driver/blob/8101fbf9ec91d952e90022606090a4be9f24aace/core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodec.java#L69). Let me know if this needs to be changed.

Reply:
Ok. If this is due to specific constraint like API of encoders not allowing encoding into existing `ByteBuffer` (but always allocating new one), then it is what it is. I was thinking encoder should be able to just add to given ByteBuffer but I guess that's not the case.

I agree it's good to have code similar to the original and they probably had a good reason to do it that way, even if it appears pretty wasteful.

So no need to change it, will approve.